modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/PlayerHeaderFragment.kt

```diff
@@ -262,6 +254,46 @@ class PlayerHeaderFragment : BaseFragment(), PlayerClickListener {
                 theme = headerViewModel.theme,
             )
         }
+
+        viewLifecycleOwner.lifecycleScope.launch {
+            viewLifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
+                viewModel.snackbarMessages.collect { message ->
+                    if (message == PlayerViewModel.SnackbarMessage.ShowBatteryWarningIfAppropriate) {
+                        warningsHelper.showBatteryWarningSnackbarIfAppropriate(snackbarParentView = view)
+                    }
+                }
+            }
+        }
+
+        viewLifecycleOwner.lifecycleScope.launch {
+            viewLifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
+                viewModel.navigationState.collect { navigationState ->
+                    when (navigationState) {
+                        is PlayerViewModel.NavigationState.ShowStreamingWarningDialog -> {
+                            warningsHelper.streamingWarningDialog(episode = navigationState.episode, snackbarParentView = view, sourceView = sourceView)
+                                .show(parentFragmentManager, "streaming dialog")
+                        }
+
+                        PlayerViewModel.NavigationState.ShowSkipForwardLongPressOptionsDialog -> {
+                            LongPressOptionsFragment().show(parentFragmentManager, "longpressoptions")
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private fun setupPlayerControlsComposeView() {
+        binding?.playerControlsComposeView?.apply {
+            setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+            setContent {
+                AppTheme(theme.activeTheme) {
+                    PlayerControls(
+                        playerViewModel = viewModel,
+                    )
+                }
+            }
+        }
     }
```

Comment:
You don't need to look at anything for this comment but I have seen the following pattern used in the code recently. It would be good to discuss if we should use this for a pattern in the app.

```
    private fun AdapterPlayerHeaderBinding.setupPlayerControlsComposeView() {
        playerControlsComposeView.setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
        playerControlsComposeView.setContent {
            AppTheme(theme.activeTheme) {
                PlayerControls(
                    playerViewModel = viewModel,
                )
            }
        }
    }
```